zzk/service/lock_test.go

```diff
@@ -118,7 +118,24 @@ func (t *ZZKTest) TestEnsureServiceLock(c *C) {
 
 	// Release lock
 	close(exit)
-	runtime.Gosched()
+
```

Comment:
The code below is supposed to be a more robust replacement for that. We should probably get John Crocker to take a look at it and make sure it's okay; it's his test.